content/post/elm-at-rakuten/index.md

```diff
@@ -599,74 +737,105 @@ b = a + 2
-a = a + 2;
-a = a * 3; // Gives 6
+[^order-doesnt-matter-ellie]:
+    [The Elm code example is available here](https://ellie-app.com/bS3bK8gnW8na1). Another situation where the order matter is when mutability is in action, for example{{< highlight javascript >}}a = 1;
+    a = a + 2;
+    a = a \* 3; // Gives 9
 
-// While
+// while, inverting the order of
+// the 2nd and 3rd line:
 
 a = 1;
-a = a * 3;
-a = a + 2; // Gives 5{{< /highlight >}}These constructs don't compile in Elm because all data is immutable. By the way, have you ever noted that with mutability, the `=` sign loses its mathematical meaning? `a = a + 2` is an impossible construct in mathematics. With immutability, it is still holding because you can only write `newA = a + 2`.
+a = a \* 3;
+a = a + 2; // Gives 5{{< /highlight >}}These constructs don't compile in Elm because all data is immutable. By the way, have you ever noted that with mutability, the `=` sign loses its mathematical meaning? `a = a + 2` is an impossible construct in mathematics. With immutability, it is still holding because you can only write `newA = a + 2`. More about this in the [Hints for Bad Recursion](https://github.com/elm/compiler/blob/master/hints/bad-recursion.md) article.
 
-Even if the two lines seem in the wrong order, it works well in Elm, but the same code will throw the error “*b* is not defined” in imperative languages.[^order-doesnt-matter-jsfiddle]
+Even if the two lines seem in the wrong order, it works well in Elm, but the same code will throw the error “_b_ is not defined” in imperative languages.[^order-doesnt-matter-jsfiddle]
 
 [^order-doesnt-matter-jsfiddle]: [Live example in JavaScript](https://jsfiddle.net/xsbhLm9z/)
 
-Refactoring became simpler because we can shuffle pieces of code around without warring.
+Refactoring became simpler because **we can shuffle pieces of code around** without warring.
 
 In our largest project, we are in the middle of a third major refactoring iteration, and we have pieces of code that are still at the first iteration and some code in the second iteration. And all of our code works well together. We are now incrementally moving all our code toward the third iteration. In Elm, you don't need to get things right from the beginning.
 
-## 16. Hiring
+[Index](#index) | [Top](#top)
+
+---
+
+## 17. Hiring
+
+Hiring for a language that is not mainstream has some downsides. For example, not many developers probably know it fluently.
 
-Hiring for a language that is not mainstream has some downsides. For example, not many developers are already fluent in it.
+However, the feeling, from engaging in the Elm slack channel, is that there are probably now more developers that know Elm compared to the number of available Elm job positions, making it possible to find developers that already master it.
```

Comment:
It feels a bit like what somebody would say about COBOL (lots of starving devs and no jobs). Is that something we love about Elm?

Reply:
I removed this paragraph